CMakeLists.txt

```diff
@@ -34,6 +34,15 @@ if(CONFIG_BOARD_NRF52_BSIM)
   zephyr_library_compile_definitions(NO_POSIX_CHEATS)
 
   file(GLOB_RECURSE HW_MODEL_SRCS . src/*.c)
+  if (CONFIG_BOARD_NRF52_BSIM)
```

Comment:
I believe we can replace `CONFIG_BOARD_NRF52_BSIM` with `CONFIG_HAS_HW_NRF_DPPIC`